### DISCUSSION TITLE
Interval to update translations

### DISCUSSION THREAD

**Discussion by Vissie2** [crowdin/ota-client-js#76] [Ideas]
Hello!

I'm creating an app using this library and came to notice there is no way to configure the automatic updating of translations for a long-living application. In the Android SDK, I believe this can be achieved by setting the `withUpdateInterval` property. This wonders me why it has not been implemented yet in this libary.

Is this a planned feature or not implemented on purpose? 

Thanks!

**Comment by andrii-bodnar** [comment]
Hi @Vissie2!

The Crowdin mobile SDKs and OTA client follow slightly different concepts in terms of translation caching. The mobile SDKs encapsulate everything a bit more than the OTA client. They store the downloaded translations in the local memory of the device, which allows to use these translations later without making requests to the distribution until the `withUpdateInterval` expires. Also, the mobile SDKs don't provide much control for the developer to request new translations - this is hidden within the SDK.

The OTA client is more platform-agnostic and gives more flexibility to the developer. It only has the in-memory cache and it's up to the developer how to implement the translation fetching flow as well as some long-term caching.

**Comment by Vissie2** [comment]
Thanks for your reply! After second thought, this does make sense and may not be the concern of this library. In our implementation we are using i18next so we would need to create a plugin or some sort to support it.
